src/morse/index.js

```diff
@@ -58,6 +58,24 @@ const MORSE_CODE = {
 
 Object.freeze(MORSE_CODE);
 
-function morse(text) {}
+function morse(text) {
+  if (text === "") return "";
+  if (!text || !text.length) throw Error("Please provide a morse string");
+  let decode = "";
+  const result = MORSE_CODE[text];
+  if (result) return result;
+  const [symbols] = text.trim().match(/\W+/);
+  const splitSymbols = symbols.split(" ");
+  splitSymbols.forEach((value, index) => {
+    if (!MORSE_CODE[value]) {
+      if (splitSymbols[index] === "" && splitSymbols[index + 1] === "") {
```

Comment:
Reading out of bounds is not encouraged. You should do a check.

Reply:
Done 👍